### DISCUSSION TITLE
Liberty Music App to play the Songs Offline & in SYNC with all other devices

### DISCUSSION THREAD

**Discussion by barayeazadi3** [WeAreMahsaAmini/Core#411] [Ideas]
# Problem

Shervin's Song Playback with high volume was limited bcs:

1. multiple devices can't play it in sync without being connected either through radio-like audio source (youtube, etc,... ), or with even apps for this particular purpose requires connecting of multiple devices in an area or across internet
2. some people like to join the movement to play it loud but are afraid to get noticed, but if it was already playing loud and they can add to the volume it reduces the chances that they being noticed

# Solution

Similar to how Authenticator apps work using a time-synced based approach across not connected devices. By considering that all smartphone and devices have synced-up accurate times we can have an playing Shervin's song like a radio/streaming song that plays in sync in any device completely offline.

We could update the app with offline songs (right now there are two great songs, Shervin's song and the new Woman's Anthem song)

_Wild Fantasies: Imagine if we could somehow hack all mosque across Iran to play that song in highest volume synchronously_

**Comment by SamanZargarzadeh** [comment]
I was looking for a keyword of "Radio" and I found your idea which is similar to what I was thinking about. I was considering that if we can have radio freq. then everybody could get it in their car and also in their home. But I don't have any knowledge in this field I don't know how we can have our own radio freq. 
your solution is great too but I think downloading a new app is very hard for people these days.
There is a simple solution to just contact with Radio Farda team and ask them to dedicate a specific wave only for playing Iran revolution music 24/7.

**Comment by SamanZargarzadeh** [comment]
Of course, if we could not have our own radio wave freq. the best idea would be PWA But Unfortunately I don't have knowledge in designing it so I will ask in the discord maybe somebody could join to this discussion.

**Comment by arashi-dev** [comment]


**Reply by SamanZargarzadeh** [reply]
Thank you very much for taking part. I believe our first goal should be to create an offline product that is simple to share in the midst of a protest. I first suggest radio frequency because I believe the car can simply tell the next car a number like AM 1539 and they will become synced, and it will spread like a virus among all cars. (I am aware that the radio has a jamming problem as well), but the most important feature that it should have is the ease of access.

1. I believe it should be more than 1 because people in the car should not be exhausted. We can start with 3 (Baraye, Sorood Zan, and Yare Dabestani) and in the next release, if it is possible, we can have more. 
2. these three songs should be one repeat 24/7, so while you open the app, it may be in the middle of a song, but it's not important as long as you sync with others.
3. Not for the initial version. I think just a blank page with a sound on and off button is enough.

**Reply by SamanZargarzadeh** [reply]
Thank you kindly. I have some product development experience, so if you need a consensus, just let me know and I'll be glad to assist.

**Reply by SamanZargarzadeh** [reply]
> 

https://twitter.com/callm3oldF/status/1578780338545135617?s=20&t=khuau7P59wiBxv4jI18iEA

Take a look at this tweet! This Twitter account creates a similar website (Radio Zan Zendegi Azadi), but it has two major differences: it plays different music (not the popular ones), and it does not work offline. We could contact them, and you could assist them in making their product work offline rather than reinventing the wheel.

**Reply by SamanZargarzadeh** [reply]
Thanks, Arash! I don't know about the technical detail, I just want to inform you about a similar product. We will wait for your version.

**Reply by radiozza** [reply]
hi
we are Radio ZZA, we look forward to collaborate if you are interested.

**Comment by arashi-dev** [comment]


**Reply by SamanZargarzadeh** [reply]
> Good news, Bad news. The initial version of the PWA website is ready and waiting for your response (supports only `baraye` song for now) 
> 
> [https://sync-song-mahsa-amini.vercel.app/](https://sync-song-mahsa-amini.vercel.app/)
> 
> 
> 
> ✅ Pros:
> 
> 1. it works offline
> 
> 2. supports dark mode
> 
> 3. supports multi-language lyrics
> 
> 4. can be deployed on static server
> 
> 5. responsive
> 
> 
> 
> ❌ Cons:
> 
> 1. **not very sync**
> 
>     devices' local time are not atomic, so there is about 1 second (maybe more, maybe less) difference. 
> 
> 2. users need a chromium-based browser (e.g. Chrome, Brave, Microsoft Edge)
> 
> 3. if we add music, the song's timeline will be updated and offline users may not get updated. so we cannot add songs frequently (lyrics are fine)
> 
> 
> 
> I will wait for your responses and if @syavash , @ramyareye agree. I will share the source

Hi Arash, Well done! I think its advantages are bolder than the cons. You put a lot of detail into it, and I appreciate it.

And about the cons:
1. we can ask our friends to give it a try and give us some feedback. 
2. I have safari, and I opened it works. 
3. I think now the app is good enough for release with "Baraye song" but I have a question if you add music and they become connected to the internet, will it update automatically? Or should they delete this PWA and get it again to become updated? 

BTW, thanks a lot for making this in a short time.
